src/app/core/services/tag-info.service.ts

```diff
@@ -1,75 +1,112 @@
 import { Injectable } from '@angular/core';
-import { HttpClient, HttpErrorResponse } from '@angular/common/http';
-import { Observable, throwError } from 'rxjs';
-import { catchError, timeout, retry, finalize } from 'rxjs/operators';
-import { SelectizeOption } from '../../shared/interfaces/selectize-option.interface';
-import { environment } from '../../../environments/environment'
+import { HttpClient } from '@angular/common/http';
+import { ServiceCacheInterceptor } from './service-cache-interceptor';
+import { environment } from '../../../environments/environment';
+import { OperatorFunction } from 'rxjs';
+import { reduce } from 'rxjs/operators';
 
 @Injectable({
   providedIn: 'root',
 })
+export class TagInfoService {
+  /* CACHE PROPS */
+  public static POPULAR_TAGS = '/api/4/tags/popular';
+  public static POPULAR_KEYS = '/api/4/keys/all';
+  public static TAG_VALUES = 'tagValues';
+  public static TAG_COMBINATIONS = 'tagCombinations';
+  public static KEY_COMBINATIONS = 'keyCombinations';
 
+  /* URL BUILDERS/STRINGS */
+  static POPULAR_TAGS_URL = `${environment.taginfo}tags/popular?page=1&rp=20&sortname=count_all&sortorder=desc`;
+  static POPULAR_KEYS_URL = `${environment.taginfo}keys/all?page=1&rp=20&filter=in_wiki&sortname=count_all&sortorder=desc`
+  /**
+   * Builds url for getting popular values for a given key
+   * @param key {string} values' key
+   * @return {string} tag values url...
+   */
+  static tagValuesUrl(key: string): string {
+    return `${environment.taginfo}key/values?key=${key}&page=1&rp=50&sortname=count_ways&sortorder=desc`;
+  }
 
-export class TagInfoService {
- 
-  tagInfoUrl: string;
-  keysMap : Map<number, any> = new Map<number, any>();
-  comboMap: Map<number, any> = new Map<number, any>();
-  popularKeys: SelectizeOption[];
-  popularTagsRequest: any;
+  /**
+   * Builds url for getting specific tag combinations
+   * @param key {string} combo key
+   * @param value {string} combo value
+   * @return {string} tag combinations url string...
+   */
+  static tagCombinationsUrl(key: string, value: string): string {
+    return `${environment.taginfo}tag/combinations?key=${key}&value=${value}&page=1&rp=50&sortname=together_count&sortorder=desc`;
+  }
 
-  constructor(private http: HttpClient) {
-    this.setTagInfoUrl();
-    this.getPopularKeyOptions();
+  /**
+   * Builds url for getting specific key combinations
+   * @param key {string} combo key
+   * @param value {string} combo value
+   * @return {string} key combinations url string...
+   */
+  static keyCombinationsUrl(key: string, value: string): string {
+    return `${environment.taginfo}key/combinations?key=${key}&value=${value}&page=1&rp=50&sortname=together_count&sortorder=desc`;
   }
 
-  setTagInfoUrl(){
-    this.tagInfoUrl = environment.taginfo;
+  static reducer(nameKey: string, valKey: string): OperatorFunction<any, any> {
```

Comment:
each of the http request wrapper functions use this to return responses serializes as in the format the comboboxes understand... 

the nameKey/valKey were figured out by looking at taginfo api docs.